src/azure-cli/azure/cli/command_modules/backup/custom_wl.py

```diff
@@ -412,6 +412,28 @@ def disable_protection(cmd, client, resource_group_name, vault_name, item, delet
     return cust_help.track_backup_job(cmd.cli_ctx, result, vault_name, resource_group_name)
 
 
+def undelete_protection(cmd, client, resource_group_name, vault_name, item):
+    container_uri = cust_help.get_protection_container_uri_from_id(item.id)
+    item_uri = cust_help.get_protected_item_uri_from_id(item.id)
+
+    backup_item_type = item_uri.split(';')[0]
+    if not cust_help.is_sql(backup_item_type) and not cust_help.is_hana(backup_item_type):
+        raise CLIError(
```

Comment:
Can you use a detailed error type based on [error handling guide](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#available-error-types)?

Reply:
Should I use ValidationError here?